### ISSUE TITLE
Issue rendering polygons that cover poles

### ISSUE THREAD

**Issue by jdgarrett** [mapbox/mapbox-gl-js#7929]
I looked through some of the existing issues related to polygon rendering and didn't see one that mentioned the issue I am seeing.

I am trying to render a large GeoJSON polygon that envelops the north pole.  Mapbox GL renders the boundaries of the polygon correctly, but any area north of the boundary is not filled.

Here is a JSFiddle that modifies the example for drawing a GeoJSON polygon:
https://jsfiddle.net/nrgy3f0o/

**Comment by jdgarrett** [comment]
I guess this is more of an issue with the polygon crossing over the dateline.  I will try to find another solution for generating the polygon.

**Comment by mourner** [comment]
The dateline crossing should be handled properly in GL JS. I think the original GeoJSON here is to blame — the maximum latitude it has is 70.94 degrees, while Web Mercator projection is limited by at least ~85.

**Comment by kerel-fs** [comment]
For reference:
This problem can be fixed by adding the two points [a, 90.0], [b, 90.0] to the Polygon at the position where the dateline is crossed, i.e. at a=min(longitudes), b=max(longitudes).

---

## Example

### broken
<img src="https://user-images.githubusercontent.com/10089607/60986398-ab141d00-a33f-11e9-9540-24005cbfc3e6.png" width="500">

```geojson
{
  "type": "FeatureCollection",
  "features": [
  {
    "type": "Feature",
    "properties":{},
    "geometry":
    {
      "type": "Polygon",
      "coordinates": [[
[-180,70],
[-20,40],
[20,40],
[180,70],
[-180,70]
      ]]
    }
  }]
}
```


### fixed
<img src="https://user-images.githubusercontent.com/10089607/60986372-9e8fc480-a33f-11e9-90c3-05217db95ba4.png" width="500">

```geojson
{
  "type": "FeatureCollection",
  "features": [
  {
    "type": "Feature",
    "properties":{},
    "geometry":
    {
      "type": "Polygon",
      "coordinates": [[
[-180,70],
[-20,40],
[20,40],
[180,70],
[180,90],
[-180,90],
[-180,70]
      ]]
    }
  }]
}
```
Added points here: `[180,90],[-180,90]`
